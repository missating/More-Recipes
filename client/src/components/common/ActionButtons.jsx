import React from 'react';
import PropTypes from 'prop-types';


const ActionButtons = props => (
  <div className="container">

    <button className="btn disabled">
      <span>
        <i className="far fa-thumbs-up" />
      </span>
      {props.singleRecipe.downvote}
    </button>

    <button className="btn disabled">
      <span>
        <i className="fa fa-thumbs-down" />
      </span>
      {props.singleRecipe.upvote}
    </button>

    <button className="btn disabled">
      <span>
        <i className="far fa-heart" />
      </span>
      {props.singleRecipe.favourite}
    </button>
  </div>
);

ActionButtons.propTypes = {
  singleRecipe: PropTypes.shape({
    favourite: PropTypes.number,
    upvote: PropTypes.number,
    downvote: PropTypes.number
  }).isRequired
};

export default ActionButtons;

