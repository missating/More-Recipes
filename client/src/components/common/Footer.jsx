import React from 'react';

const Footer = () => (
  <footer>
    <hr />
    <div className="container bg-color">
      <div className="row">

        <div className="col-md-12">
          <div className="text-center">

            <h5 style={{ color: '#333' }}>
              Vanessa Ating &copy;&nbsp;
              {new Date().getFullYear()}
            </h5>

          </div>
        </div>
      </div>
    </div>

  </footer>
);

export default Footer;
