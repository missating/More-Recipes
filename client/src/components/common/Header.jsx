import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// actions
import signUserOut from '../../actions/signout';

export const Header = props => (
  <header>
    <nav
      className="navbar navbar-expand-md navbar-dark fixed-top bg-dark bg-black"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        More Recipes
      </a>
      <button
        className="navbar-toggler navbar-toggler-right"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNavDropdown"
        aria-controls="navbarNavDropdown"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navbarNavDropdown">
        <ul className="navbar-nav ml-auto">
          <span className="sr-only">(current)</span>
          <li className="nav-item">
            <Link className="nav-link" to="/recipes">Recipes</Link>
          </li>

          {
            props.authenticated &&
            <Link className="nav-link" to="/recipe/add">Create Recipe</Link>
          }

          {
            props.authenticated &&
            <li className="nav-item dropdown user-name">
              <a
                className="nav-link dropdown-toggle"
                id="navbarDropdownMenuLink"
                data-toggle="dropdown"
                aria-haspopup="true"
                aria-expanded="false"
              >
                {
                  props.authenticated &&
                  <span>Hi, {props.user.fullname}</span>
                }

              </a>
              <div
                className="dropdown-menu"
                aria-labelledby="navbarDropdownMenuLink"
              >
                <Link
                  className="nav-link"
                  to="/profile"
                >
                  My Profile
                </Link>
                <Link
                  className="nav-link"
                  to="/users/recipes"
                >
                  My Recipes
                </Link>
                <Link
                  className="nav-link"
                  to="/users/favourites"
                >
                  My Favourites
                </Link>
              </div>
            </li>
          }

          {
            props.authenticated &&
            <li className="nav-item">
              <button
                className="btn-secondary"
                style={{ width: '100px' }}
                id="signout"
                onClick={() => { props.signUserOut(); }}
              >
                SIGN OUT
              </button>
            </li>
          }

        </ul>
      </div>
    </nav>

  </header>
);

Header.propTypes = {
  authenticated: PropTypes.bool.isRequired,
  signUserOut: PropTypes.func.isRequired,
  user: PropTypes.shape({
    fullname: PropTypes.string
  }),
};

Header.defaultProps = {
  user: {}
};

const mapStateToProps = state => ({
  authenticated: state.auth.isAuthenticated,
  user: state.auth.user
});

const mapDispatchToProps = dispatch => ({
  signUserOut: () => {
    dispatch(signUserOut());
  }
});

const ConnectedComponent =
  connect(mapStateToProps, mapDispatchToProps)(Header);
export { ConnectedComponent as ConnectedHeader };

