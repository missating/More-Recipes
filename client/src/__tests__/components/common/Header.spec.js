import React from 'react';
import toJSON from 'enzyme-to-json';
import { Header } from '../../../components/common/Header';

describe('Header component', () => {
  const props = {
    authenticated: true,
    user: {
      fullname: 'test'
    },
    signUserOut: jest.fn(() => Promise.resolve())
  };

  it('Should render', () => {
    const wrapper = shallow(<Header
      {...props}
    />);
    expect(wrapper.find('nav').length).toEqual(1);
    expect(wrapper.find('div').length).toEqual(2);
    expect(wrapper.find('li').length).toEqual(3);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Should sign user out when button is clicked', () => {
    const wrapper = shallow(<Header
      {...props}
    />);
    wrapper.find('#signout').simulate('click');
  });
});

