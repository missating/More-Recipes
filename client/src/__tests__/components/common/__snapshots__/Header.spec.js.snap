// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render 1`] = `
<header>
  <nav
    className="navbar navbar-expand-md navbar-dark fixed-top bg-dark bg-black"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      More Recipes
    </a>
    <button
      aria-controls="navbarNavDropdown"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbarNavDropdown"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNavDropdown"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="sr-only"
        >
          (current)
        </span>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/recipes"
          >
            Recipes
          </Link>
        </li>
        <Link
          className="nav-link"
          replace={false}
          to="/recipe/add"
        >
          Create Recipe
        </Link>
        <li
          className="nav-item dropdown user-name"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            className="nav-link dropdown-toggle"
            data-toggle="dropdown"
            id="navbarDropdownMenuLink"
          >
            <span>
              Hi, 
              test
            </span>
          </a>
          <div
            aria-labelledby="navbarDropdownMenuLink"
            className="dropdown-menu"
          >
            <Link
              className="nav-link"
              replace={false}
              to="/profile"
            >
              My Profile
            </Link>
            <Link
              className="nav-link"
              replace={false}
              to="/users/recipes"
            >
              My Recipes
            </Link>
            <Link
              className="nav-link"
              replace={false}
              to="/users/favourites"
            >
              My Favourites
            </Link>
          </div>
        </li>
        <li
          className="nav-item"
        >
          <button
            className="btn-secondary"
            id="signout"
            onClick={[Function]}
            style={
              Object {
                "width": "100px",
              }
            }
          >
            SIGN OUT
          </button>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
